android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -144,31 +236,294 @@ private String getCurrentCountry() {
+    TelephonyManager telephonyManager =
+            (TelephonyManager) reactContext.getSystemService(Context.TELEPHONY_SERVICE);
+
+    if (ActivityCompat.checkSelfPermission(reactContext, Manifest.permission.READ_PHONE_STATE) != PackageManager.PERMISSION_GRANTED) {
+      @SuppressLint("HardwareIds") String deviceId = telephonyManager.getDeviceId();
+
+      for (String known_deviceId : DEVICE_IDS) {
+        if (known_deviceId.equalsIgnoreCase(deviceId)) {
+          return true;
+        }
 
+      }
+    }
+
+    return false;
+  }
+
+  private boolean checkTelephony() {
+    if (ContextCompat.checkSelfPermission(reactContext, Manifest.permission.READ_PHONE_STATE)
+            == PackageManager.PERMISSION_GRANTED && this.isTelephony && isSupportTelePhony()) {
+      return checkPhoneNumber()
+              || checkDeviceId()
+              || checkImsi()
+              || checkOperatorNameAndroid();
+    }
+    return false;
+  }
+
+  private boolean checkQEmuDrivers() {
+    for (File drivers_file : new File[]{new File("/proc/tty/drivers"), new File("/proc/cpuinfo")}) {
+      if (drivers_file.exists() && drivers_file.canRead()) {
+        byte[] data = new byte[1024];
+        try {
+          InputStream is = new FileInputStream(drivers_file);
+          is.read(data);
+          is.close();
+        } catch (Exception exception) {
+          exception.printStackTrace();
+        }
+
+        String driver_data = new String(data);
+        for (String known_qemu_driver : QEMU_DRIVERS) {
+          if (driver_data.contains(known_qemu_driver)) {
+            return true;
+          }
+        }
+      }
+    }
+
+    return false;
+  }
+
+  private boolean checkFiles(String[] targets, String type) {
+    for (String pipe : targets) {
+      File qemu_file = new File(pipe);
+      if (qemu_file.exists()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private String getProp(Context context, String property) {
+    try {
+      ClassLoader classLoader = context.getClassLoader();
+      Class<?> systemProperties = classLoader.loadClass("android.os.SystemProperties");
+
+      Method get = systemProperties.getMethod("get", String.class);
+
+      Object[] params = new Object[1];
+      params[0] = property;
+
+      return (String) get.invoke(systemProperties, params);
+    } catch (Exception exception) {
+      // empty catch
+    }
+    return null;
+  }
+
+  private boolean checkQEmuProps() {
+    int found_props = 0;
+
+    for (Property property : PROPERTIES) {
+      String property_value = getProp(reactContext, property.name);
+      if ((property.seek_value == null) && (property_value != null)) {
+        found_props++;
+      }
+      if ((property.seek_value != null)
+              && (property_value.contains(property.seek_value))) {
+        found_props++;
+      }
+
+    }
+
+    if (found_props >= MIN_PROPERTIES_THRESHOLD) {
+      return true;
+    }
+    return false;
+  }
+
+  private boolean checkIp() {
```

Comment:
I don't understand why we spawn a unix process to run netcfg to get the IP, instead of just ... getting the IP address from Java? This module already loads the IP address if it's possible - is this method somehow more robust? If so you need to document exactly why with a link to supporting documentation

Reply:
I also considering this method, the objective was to identify _eth_, _tunl_ or _wlan_... connections related to some emulators, but as I said I'm considering if we should keep it or stay only with **checkEth0Interface** method. Maybe I could remove without loose precision on detecting an emulator device.

Reply:
removed 
https://github.com/react-native-community/react-native-device-info/pull/721/commits/da3c60a87e5b883e385f4a5d8d43092f34e78b52